samples/language/entity-extraction/create-dataset.v1beta1.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2019, Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+`use strict`;
+async function main(
+  projectId = 'YOUR_PROJECT_ID',
+  computeRegion = 'YOUR_REGION_NAME',
+  datasetName = 'YOUR_DATASET_NAME'
+) {
+  // [START automl_natural_language_entity_create_dataset]
+  const automl = require(`@google-cloud/automl`);
+  const util = require(`util`);
+  const client = new automl.v1beta1.AutoMlClient();
+
+  /**
+   * Demonstrates using the AutoML client to create a dataset
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = '[PROJECT_ID]' e.g., "my-gcloud-project";
+  // const computeRegion = '[REGION_NAME]' e.g., "us-central1";
+  // const datasetName = '[DATASET_NAME]' e.g., "myDataset";
+
+  // A resource that represents Google Cloud Platform location.
+  const projectLocation = client.locationPath(projectId, computeRegion);
+
+  // Set dataset name and metadata.
+  const myDataset = {
+    displayName: datasetName,
+    textExtractionDatasetMetadata: {},
+  };
+
+  // Create a dataset with the dataset metadata in the region.
+  client
```

Comment:
so I think we're now doing a model that looks more like this:
https://github.com/googleapis/nodejs-translate/pull/234/files

The `main` function can be synchronous, and have an async method in it called `createDataset`.  Check out the other samples in that PR, and we should be pretty consistent!